src/riverroute/RtmMod.F90

```diff
@@ -1,2844 +1,2407 @@
+      real(r8),allocatable :: factorList(:)
+      integer ,allocatable :: factorIndexList(:,:)
+      integer              :: srcTermProcessing_Value = 0
+      character(len=*),parameter :: FORMI = '(2A,2i10)'
+      character(len=*),parameter :: FORMR = '(2A,2g15.7)'
+      character(len=*),parameter :: subname = '(MOSART_init2)'
+      !-----------------------------------------------------------------------
+
+      rc = ESMF_SUCCESS
+
+      ! Set up pointer arrays into srcfield and dstfield
+      call ESMF_FieldGet(srcfield, farrayPtr=src_direct, rc=rc)
+      if (chkerr(rc,__LINE__,u_FILE_u)) return
+      call ESMF_FieldGet(dstfield, farrayPtr=dst_direct, rc=rc)
+      if (chkerr(rc,__LINE__,u_FILE_u)) return
+      src_direct(:,:) = 0._r8
+      dst_direct(:,:) = 0._r8
+
+      ! Calculate map for direct to outlet mapping
+      ! The route handle rh_direct will then be used in MOSART_run
+      cnt = rtmCTL%endr - rtmCTL%begr + 1
+      allocate(factorList(cnt))
+      allocate(factorIndexList(2,cnt))
+      cnt = 0
+      do nr = rtmCTL%begr,rtmCTL%endr
+         cnt = cnt + 1
+         if (rtmCTL%outletg(nr) > 0) then
+            factorList(cnt) = 1.0_r8
+            factorIndexList(1,cnt) = rtmCTL%gindex(nr)
+            factorIndexList(2,cnt) = rtmCTL%outletg(nr)
+         else
+            factorList(cnt) = 1.0_r8
+            factorIndexList(1,cnt) = rtmCTL%gindex(nr)
+            factorIndexList(2,cnt) = rtmCTL%gindex(nr)
+         endif
+      enddo
+
+      call ESMF_FieldSMMStore(srcField, dstField, rh_direct, factorList, factorIndexList, &
+           ignoreUnmatchedIndices=.true., srcTermProcessing=srcTermProcessing_Value, rc=rc)
+      if (chkerr(rc,__LINE__,u_FILE_u)) return
+
+      deallocate(factorList)
+      deallocate(factorIndexList)
+
+      if (mainproc) write(iulog,*) subname," Done initializing rh_direct "
+
+      ! ---------------------------------------
+      ! Read in data from frivinp_rtm
+      ! ---------------------------------------
+
+      begr = rtmCTL%begr
+      endr = rtmCTL%endr
+
+      if(endr >= begr) then
+
+         ! routing parameters
+         call ncd_pio_openfile (ncid, trim(frivinp_rtm), 0)
+         call pio_seterrorhandling(ncid, PIO_INTERNAL_ERROR)
+
+         allocate(compdof(rtmCTL%lnumr))
+         cnt = 0
+         do n = rtmCTL%begr,rtmCTL%endr
+            cnt = cnt + 1
+            compDOF(cnt) = rtmCTL%gindex(n)
+         enddo
+
+         ! setup iodesc based on frac dids
+         ier = pio_inq_varid(ncid, name='frac', vardesc=vardesc)
+         ier = pio_inq_vardimid(ncid, vardesc, dids)
+         ier = pio_inq_dimlen(ncid, dids(1),dsizes(1))
+         ier = pio_inq_dimlen(ncid, dids(2),dsizes(2))
+         call pio_initdecomp(pio_subsystem, pio_double, dsizes, compDOF, iodesc_dbl)
+         call pio_initdecomp(pio_subsystem, pio_int   , dsizes, compDOF, iodesc_int)
+         deallocate(compdof)
+         call pio_seterrorhandling(ncid, PIO_BCAST_ERROR)
```

Comment:
It's not clear to me why we are using PIO_BCAST_ERROR here as I see no error handling in the following code.

Reply:
Done.